src/main/java/uk/gov/hmcts/reform/idam/health/vault/VaultEnvironmentPostProcessor.java

```diff
@@ -25,8 +24,10 @@
     protected static final String VAULT_BASE_URL = "azure.keyvault.uri";
     protected static final String VAULT_CLIENT_ID = "azure.keyvault.client-id";
     protected static final String VAULT_CLIENT_KEY = "azure.keyvault.client-key";
+    protected static final String VAULT_CREDENTIAL_TYPE = "azure.keyvault.credential.type";
```

Comment:
If we decide not to use this "switch" then we need to be careful that we don't build a default client the first time spring calls the post processor (when there are no properties in the environment at all)

Reply:
Maybe it's not a big deal even if we do _as long as_ we have the option to overwrite it later (when there are properties in the environment).

Reply:
Removed vault credential type as a switch for client type - use presence of azure.keyvault.msi.url property instead.